src/audio/dai-zephyr.c

```diff
@@ -1537,13 +1536,6 @@ int dai_common_copy(struct dai_data *dd, struct comp_dev *dev, pcm_converter_fun
 	avail_bytes = stat.pending_length;
 	free_bytes = stat.free;
 
-	buf_c = buffer_acquire(dd->dma_buffer);
-
-	dma_fmt = audio_stream_get_frm_fmt(&buf_c->stream);
-	sampling = get_sample_bytes(dma_fmt);
-
-	buffer_release(buf_c);
-
 	/* handle module runtime unbind */
```

Comment:
Is it guaranteed dd->sampling matches with the "dma_buffer" fmt?
UPDATE: ack, so you add dd->samping in this PR, right, so this is good.

Reply:
yes, dd->sampling should keep unchanged with same dd.